common/realtime.py

```diff
@@ -52,10 +52,12 @@ def __init__(self, rate: float, print_delay_threshold: float | None = 0.0) -> No
     self._frame = 0
     self._remaining = 0.0
     self._process_name = getproctitle()
-    self._dts = deque([self._interval], maxlen=100)
     self._last_monitor_time = -1.
     self._next_frame_time = -1.
 
+    self.avg_dt = MovingAverage(100)
+    self.avg_dt.add_value(self._interval)
```

Comment:
why do we need to add a value every time?

Reply:
Add an initial value in `__init__` because `ratekeeper` calls `lagging()` before it updates with `monitor_time()`. This ensures the first call to lagging() returns a valid average, avoiding uninitialized values, similar to how `self._dts = deque([self._interval], maxlen=100)` works